### PR TITLE
Make the TP importer more configurable

### PR DESCRIPTION
**Pull request by dwhenry**
This is required to import the initial monthly
email which has a different subject, filename and
sheet name compared to the daily emails.

### CODE REVIEW

File: spec/features/import_tp_call_data_spec.rb [hunk 1]
```diff
@@ -5,10 +5,14 @@
 RSpec.feature 'Importing twilio call data', vcr: { cassette_name: 'twilio_single_page_of_data' } do
   let(:start_date) { Date.new(2016, 4, 11) }
   let(:end_date) { Date.new(2016, 4, 12) }
-  let(:tp_config) { double(:config, tp: { user_name: 'researchuploads@pensionwise.gob.uk', password: '1234' }) }
-
-  before do
-    allow(Rails.configuration).to receive(:x).and_return(tp_config)
+  let(:config) do
+    ActiveSupport::OrderedOptions[
```
Comment (benlovell) [inline spec/features/import_tp_call_data_spec.rb]: TIL `ActiveSupport::OrderedOptions`.

I would usually use an `OpenStruct` for the same purpose.

File: lib/importers/tp/importer.rb [hunk 2]
```diff
@@ -1,15 +1,16 @@
 module Importers
   module TP
     class Importer
-      def initialize(retriever: Retriever, call_record: CallRecord, saver: Saver)
-        @retriever = retriever.new(config: Rails.configuration.x.tp)
+      def initialize(retriever: Retriever, call_record: CallRecord, saver: Saver, config: Rails.configuration.x.tp)
+        @retriever = retriever.new(config: config)
```
Comment (benlovell) [inline lib/importers/tp/importer.rb]: It's not a pattern i'm overly-enamoured with, but using a globally scoped configuration object might win out over having to pass this `config` around several layers. Alternatively, why not just make the `Retriever`'s `config` default to `Rails.configuration.x.tp` too? So we don't need to pass it along here...

Comment (dwhenry) [inline lib/importers/tp/importer.rb]: As discuss, going to leave as is as this allows us this limits the binding to `Rails.configuration.x` as it only has a single point of entry into the code.

File: spec/features/import_tp_call_data_spec.rb (hunk 1 above)
Comment (dwhenry) [inline spec/features/import_tp_call_data_spec.rb]: Used `ActiveSupport::OrderedOptions` it what `Rails.configuration.x.tp` is. 

Using `OpenStruct` causes the tests to fail with:

```
no implicit conversion of OpenStruct into Hash
```

**Pull request merged**